pynitrokey/cli/nk3/__init__.py

```diff
@@ -172,23 +172,36 @@ def set_config(ctx: Context, key: str, value: str, force: bool, dry_run: bool) -
     """
 
     with ctx.connect_device() as device:
-        # before the confirmation prompt, check if the config value is supported
-        if not device.admin.has_config(key):
+        config_fields = device.admin.list_available_fields()
+
+        field_metadata = None
+        for field in config_fields:
+            if field.name == key:
+                field_metadata = field
+
+
+        if field_metadata is None:
+            print(
+                "Changing configuration values can have unexpected side effects, including data loss.",
+                file=sys.stderr,
+            )
+            print(
+                "This should only be used for development and testing.",
+                file=sys.stderr,
+            )
+            if not force:
+                raise CliException(
+                    "Unknown config values can only be set if the --force/-f flag is set.  Aborting.",
+                    support_hint=False,
+                )
+
+        if not force and not field_metadata.ty.is_valid(value):
             raise CliException(
-                f"The configuration option '{key}' is not supported by the device.",
+                f'Unknown config value. Expected {field_metadata.ty}, got `{value}`. Unknown config values can only be set if the --force/-f flag is set.  Aborting.',
```

Comment:
AFAIS this is the case for known config fields with invalid values:

```suggestion
                f'Invalid config value for {field}: expected {field_metadata.ty}, got `{value}`. Aborting.',
```

Reply:
Applied in bfa092b13b89b11199a3979e8adad2f1c3ff229a